plugin/core/protocol.py

```diff
@@ -720,5298 +813,5150 @@ class TokenFormat(StrEnum):
 
-FoldingRangeParams = TypedDict('FoldingRangeParams', {
-    # The text document.
-    'textDocument': 'TextDocumentIdentifier',
-    # An optional token that a server can use to report work done progress.
-    'workDoneToken': NotRequired['ProgressToken'],
-    # An optional token that a server can use to report partial results (e.g. streaming) to
-    # the client.
-    'partialResultToken': NotRequired['ProgressToken'],
-})
-""" Parameters for a {@link FoldingRangeRequest}. """
-
-
-FoldingRange = TypedDict('FoldingRange', {
-    # The zero-based start line of the range to fold. The folded area starts after the line's last character.
-    # To be valid, the end must be zero or larger and smaller than the number of lines in the document.
-    'startLine': Uint,
-    # The zero-based character offset from where the folded range starts. If not defined, defaults to the length of the start line.
-    'startCharacter': NotRequired[Uint],
-    # The zero-based end line of the range to fold. The folded area ends with the line's last character.
-    # To be valid, the end must be zero or larger and smaller than the number of lines in the document.
-    'endLine': Uint,
-    # The zero-based character offset before the folded range ends. If not defined, defaults to the length of the end line.
-    'endCharacter': NotRequired[Uint],
-    # Describes the kind of the folding range such as `comment' or 'region'. The kind
-    # is used to categorize folding ranges and used by commands like 'Fold all comments'.
-    # See {@link FoldingRangeKind} for an enumeration of standardized kinds.
-    'kind': NotRequired['FoldingRangeKind'],
-    # The text that the client should show when the specified range is
-    # collapsed. If not defined or not supported by the client, a default
-    # will be chosen by the client.
-    #
-    # @since 3.17.0
-    'collapsedText': NotRequired[str],
-})
-""" Represents a folding range. To be valid, start and end line must be bigger than zero and smaller
-than the number of lines in the document. Clients are free to ignore invalid ranges. """
+    @since 3.16.0 """
+    item: 'CallHierarchyItem'
+    workDoneToken: NotRequired['ProgressToken']
+    """ An optional token that a server can use to report work done progress. """
+    partialResultToken: NotRequired['ProgressToken']
+    """ An optional token that a server can use to report partial results (e.g. streaming) to
+    the client. """
 
 
-FoldingRangeRegistrationOptions = TypedDict('FoldingRangeRegistrationOptions', {
-    # A document selector to identify the scope of the registration. If set to null
-    # the document selector provided on the client side will be used.
-    'documentSelector': Union['DocumentSelector', None],
-    # The id used to register the request. The id can be used to deregister
-    # the request again. See also Registration#id.
-    'id': NotRequired[str],
+CallHierarchyIncomingCall = TypedDict('CallHierarchyIncomingCall', {
```

Comment:
Why is this still using `TypedDict`

Reply:
https://github.com/sublimelsp/LSP/pull/2500/files#diff-049653877a86e4c220cd537d174655150e80b4ce0077d48742cc72b4654e2112R1126

It has a property "from" which is a reserved keyword and cannot be used as a valid class property name. the generator will not use the class syntax in that case.